src/sentry/tasks/recap_servers.py

```diff
@@ -0,0 +1,195 @@
+
+    for project_id in project_ids:
+        poll_project_recap_server.delay(project_id)
+
+
+@instrumented_task(
+    name="sentry.tasks.poll_project_recap_server",
+    queue="recap_servers",
+)
+def poll_project_recap_server(project_id: int, **kwargs) -> None:
+    try:
+        project = Project.objects.get(id=project_id)
+    except Project.DoesNotExist:
+        logger.warning("Polled project do not exist", extra={"project_id": project_id})
+        return
+
+    if not features.has("projects:recap-server", project):
+        logger.info(
+            "Recap server polling feature is not enabled for a given project",
+            extra={"project_id": project_id},
+        )
+        return
+
+    recap_server_url = project.get_option(RECAP_SERVER_URL_OPTION)
+    # Just a guard in case someone removes recap url in the exact moment we trigger polling task
+    if recap_server_url is None:
+        logger.warning(
+            "Polled project has no recap server url configured", extra={"project": project}
+        )
+        return
+
+    latest_id = project.get_option(RECAP_SERVER_LATEST_ID, 0)
+    url = recap_server_url.strip().rstrip("/") + "/rest/v1/crashes;sort=id:ascending"
+
+    # For initial query, we limit number of crashes to 1_000 items, which is the default of Recap Server,
+    # and for all following requests, we do not limit the number, as it's already capped at 10_000 by the server.
+    # For non-initial queries, we want to filter for all events that happened _after_ our previously
+    # fetched crashes, base on the most recent ID.
+    if latest_id == 0:
+        url = f"{url};limit=1000"
+    else:
+        # Apache Solr format requires us to encode the query.
+        # Exclusive bounds range - {N TO *}
+        url = url + urllib.parse.quote(f";q=id:{{{latest_id} TO *}}", safe=";=:")
+
+    headers = {
+        "Accept": "application/vnd.scea.recap.crashes+json; version=1",
+    }
+    access_token = project.get_option(RECAP_SERVER_TOKEN_OPTION, None)
+    if access_token is not None:
+        headers["Authorization"] = f"Bearer {access_token}"
+
+    result = http.fetch_file(url, headers=headers)
+
+    try:
+        crashes = json.loads(result.body)
+        if not isinstance(crashes, dict):
+            logger.exception(
+                "Polled project endpoint did not responded with json object",
+                extra={"project": project},
+            )
+            return
+    except json.JSONDecodeError as exc:
+        logger.exception(
+            "Polled project endpoint did not responded with valid json",
+            exc_info=exc,
+            extra={"project": project, "url": url},
+        )
+        return
+
+    if crashes.get("results") is None or crashes.get("results") == 0:
+        return
+
+    for crash in crashes["_embedded"]["crash"]:
+        latest_id = max(latest_id, crash["id"])
+        store_crash(crash, project, url)
+
+    project.update_option(RECAP_SERVER_LATEST_ID, latest_id)
```

Comment:
to be exception safe, I would:
- wrap the loop in try/finally
- update the `latest_id` after `store_crash` returned
- do the `update_option` in a `finally` block, so it is always executed and updates the id to the last successfully processed crash when the code raises an exception.

Reply:
I was thinking about it, but it still doesn't prevent cases of missing some data when one of the JSON payloads is not handled correctly (but should).

Eg. when `id=[1,5,8]` and `5` is somehow not translated correctly, we still store `8` as the latest, as it was well, "latest", but `5` will never be retried. I think it's not that bad anyway tbh, as we'll get everything ironed out as we go, and if something is not handled, it means that it was indeed malformed (which should not happen, as recap server controls the integrity of the data).

Will make the update you mentioned anyway though, as it's a good improvement.